### ISSUE TITLE
Overflow menu: on close lost keyboard focus

### ISSUE THREAD

**Issue by diego-codes** [carbon-design-system/carbon#595]
## Detailed description

Describe in detail the issue you're having. Is this a feature request (new component, new icon), a bug, or a general issue?
I think it's a bug.

> Is this issue related to a specific component? 

Yes, the Overflow Menu

> What did you expect to happen? What happened instead? What would you like to see changed?

When using just the keyboard and I make an option selection. I expect the keyboard focus to be placed back onto the overflow trigger "⠸" button. Instead the focus stays on the selected option and because the menu is hidden away, the keyboard focus is lost from a user's perspective.

I would like the trigger "⠸" button to take back keyboard focus once the overflow menu has closed because an option has been selected.

> What browser are you working in? 

Firefox 58.0.2 (64-bit). Mac OS High Sierra.

> What version of the Carbon Design System are you using? 

8.10.1

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

Experimental Carbon Component features.

## Steps to reproduce the issue

1. <kbd>tab</kbd> to overflow menu trigger button.
2. Press <kbd>enter</kbd> to open overflow menu.
3. Make a selection with keyboard with <kbd>tab</kbd> and <kbd>enter</kbd>.
4. Overflow menu disappears and keyboard focus goes missing.

## Add labels
Please choose the appropriate label(s) from our existing label list to ensure that your issue is properly categorized. This will help us to better understand and address your issue.
